mne/annotations.py

```diff
@@ -817,10 +817,21 @@ def _select_annotations_based_on_description(descriptions, event_id, regexp):
 
 
 def _check_event_id(event_id, raw):
+    from .io.brainvision.brainvision import _BVEventParser
+    from .io.brainvision.brainvision import _check_bv_annot
+    from .io.brainvision.brainvision import RawBrainVision
+    from .io import RawFIF, RawArray
+
     if event_id is None:
         return _DefaultEventParser()
     elif event_id == 'auto':
-        return getattr(raw, '_get_auto_event_id', _DefaultEventParser)()
+        if isinstance(raw, RawBrainVision) or (
+            isinstance(raw, (RawFIF, RawArray)) and
+            _check_bv_annot(raw.annotations.description)
```

Comment:
Maybe we should log/warn in case the BV parser is called on a RawFIF/RawArray?

Reply:
I would not warn because the save-load round-trip should be a reasonable use case. But a `logger.info` seems fine